### DISCUSSION TITLE
Custom HTML, Call image from server

### DISCUSSION THREAD

**Discussion by itiloo** [rejetto/hfs#565] [Q&A] [answered]
Hi Rejetto:

I want to call an image hosted on the server to customize the fronted html. It's possible?

 it does not work.
 
![image](https://github.com/rejetto/hfs/assets/94358936/21653f90-14e7-4c97-b59a-c55b323b9326)

Thank you so much.

**Comment by rejetto** [comment]
of course, it is a normal web server, and you can include any image you want.
just be sure that the address is correct.
i cannot check that your address is correct without seeing.

**Comment by itiloo** [comment]
http://10.10.10.10/ open the frontend

html code "img src http://10.10.10.10/images/logo2.png not open image

I have in the root directory de images directory and de logo2.png inside

**Comment by rejetto** [comment]
try to open the address in the browser, directly, and see if you can see the image http://10.10.10.10/images/logo2.png

**Comment by itiloo** [comment]
![image](https://github.com/rejetto/hfs/assets/94358936/0d5e2868-3e5c-4028-8434-36e51be79b6f)

![image](https://github.com/rejetto/hfs/assets/94358936/abe7eb79-10c7-43e6-a44e-5fbcec5de5c7)

![image](https://github.com/rejetto/hfs/assets/94358936/06696a82-67fb-4725-ab19-d8c08bae9754)

**Comment by itiloo** [comment, marked as answer]
Solved Rejetto, the virtual file system was missing, thanks and sorry

**Comment by PH5HP** [comment]
Hi,

TNX for this info. I did host my images on my webspace where my PH5HP website is hosted. I managed to move the images to the HFS/images directory. Feels better here...

I noticed to make it work you have to set the property 'Who can download' of the images directory in the virtual file system to 'Anyone', because if you set it to 'No one' it seems that the images can not be accessed. Is this right? All other properties are set to 'No one'... 

Kind regards,
Henk - ph5hp.nl - fileserver.ph5hp.nl

**Reply by itiloo** [reply]
Yes, I have it like that too, if not, it doesn't work.

**Reply by rejetto** [reply]
"download" here can be misleading, because any usage like "showing" requires the browser to actually download the image first. That's why you have to give "download" to display it.
I'm not sure if saying "Who can get the file" could be any better.
Internally it's called "can_read", but this name is not showed to the user.

**Reply by PH5HP** [reply]
Hi Rejetto,

I am also seeing people getting 'Fully Downloaded' regarding the images I host locally. Is this normal or are people download the images directly? I have all options standing on 'No one' except 'Who can download'. Is this normal?

Kind regards,
Henk - ph5hp.nl - fileserver.ph5hp.nl

**Reply by rejetto** [reply]
it is normal. As I said in my previous message, if the browser is showing the image, then it downloaded it first.
HFS is not attempting to make any difference, as there's no difference from its point of view. I'm not even sure if it's possible to make a difference, but it could be interesting to not log those requests when the "Log interface loading" is disabled.

**Reply by PH5HP** [reply]
Hi Rejetto,

I think it is possible to make a difference between images and 'normal ' files which are downloaded. You could add an option to define directories which are used by the 'system' (HFS) and if it is a file from a system directory you could display somethink like 'system file' or 'system image'... whatever you want instead of displaying the same message 'fully downloaded'...

Does this make sense?

Kind regards,
Henk - ph5hp.nl - fileserver.ph5hp.nl

**Reply by rejetto** [reply]
i have an idea to make this automatic. I'm not 100% sure it will work, but may be worth a try.
Other than this, we may have an option to not-log based on the path of the file, but i would not go for it now.

**Reply by rejetto** [reply]
[0.53.0-beta7.5 hfs-windows.zip](https://github.com/rejetto/hfs/files/15476584/hfs-windows.zip)
in this version i'm attempting to automatically detect requests that should be considered part of the interface, and so comply with the "log interface" option.
Would you care having a look and see if it works well for you?

**Reply by PH5HP** [reply]
Hi Rejetto,

Testing this.

Kind regards,
Henk - ph5hp.nl - fileserver.ph5hp.nl

**Reply by PH5HP** [reply]
Hi Rejetto,

I think it works, because I did not see any messages in the log about loading images regarding my layout of HFS.

Kind regards,
Henk - ph5hp.nl - fileserver.ph5hp.nl

**Reply by PH5HP** [reply]
Hi,

I spoke to soon. In the screenshot you can see 2 images loaded by HFS. I am using the latest beta (8).

Kind regards,
Henk - ph5hp.nl - fileserver.ph5hp.nl

![image](https://github.com/rejetto/hfs/assets/140949730/cf62a813-1eda-480a-9331-62706578fdad)

**Reply by rejetto** [reply]
that's not strange on itself.
I skip the log only if the image is loaded as part of the page, but you are still allowed to load them stand-alone, and in this case they will be logged.

**Reply by PH5HP** [reply]
Hi,

I loaded them as part of the interface by accessing https://fileserver.ph5hp.nl in my browser...

Kind regards,
Henk -ph5hp.nl - fileserver.ph5hp.nl

**Reply by rejetto** [reply]
can you reproduce the problem?
